### PR TITLE
Add optional instance Alias

### PR DESCRIPTION
**Pull request by enisoc**
This is a proposed implementation of an idea from #150, to let me associate an identifier with each instance that's independent of its hostname. My main use for this will be to assign the Alias automatically from Vitess, and then have Orchestrator pass along the Alias when executing hooks, so Vitess knows which instance was promoted.

This essentially lets me translate between Orchestrator's idea of identity (host:port) and Vitess's idea of identity (tablet alias). Otherwise, if Orchestrator tells Vitess the host:port, Vitess would have to scan its list of all tablets to find the one whose MySQL has that host:port.

### CODE REVIEW

File: go/db/db.go [hunk 1]
```diff
@@ -573,6 +573,11 @@ var generateSQLPatches = []string{
 	`,
 	`
 		ALTER TABLE
+			database_instance
```
Comment (shlomi-noach) [inline go/db/db.go]: Although not strictly required in technical terms, please move this statement to the end of statements array; I prefer observing the array as a changelog, such that I can compare any two versions and have one as prefix of the other.

File: go/db/db.go [hunk 2]
```diff
@@ -611,6 +616,11 @@ var generateSQLPatches = []string{
 			ADD COLUMN slave_hosts text CHARACTER SET ascii NOT NULL
 	`,
 	`
+		ALTER TABLE
```
Comment (shlomi-noach) [inline go/db/db.go]: Likewise, please move to end of array

File: go/inst/instance_dao.go [hunk 3]
```diff
@@ -1456,7 +1462,7 @@ func ReadClustersInfo(clusterName string) ([]ClusterInfo, error) {
 		select
 			cluster_name,
 			count(*) as count_instances,
-			ifnull(min(alias), cluster_name) as alias,
+			ifnull(min(cluster_alias.alias), cluster_name) as alias,
```
Comment (shlomi-noach) [inline go/inst/instance_dao.go]: I can see this name collision is going to be confusing. Would you mind changing the column and field names to `instance_alias` and `InstanceAlias`, respectively?

File: go/inst/instance_dao.go [hunk 4]
```diff
@@ -1630,6 +1637,7 @@ func writeInstance(instance *Instance, instanceWasActuallyFound bool, lastError
 		}
 		insertQuery := fmt.Sprintf(`
     	insert %s into database_instance (
+    		alias,
```
Comment (shlomi-noach) [inline go/inst/instance_dao.go]: Please move to end of list, make it append-only

File: go/logic/topology_recovery.go [hunk 5]
```diff
@@ -156,6 +158,9 @@ func replaceCommandPlaceholders(command string, topologyRecovery *TopologyRecove
 	if topologyRecovery.SuccessorKey != nil {
 		command = strings.Replace(command, "{successorHost}", topologyRecovery.SuccessorKey.Hostname, -1)
 		command = strings.Replace(command, "{successorPort}", fmt.Sprintf("%d", topologyRecovery.SuccessorKey.Port), -1)
+		// As long as SucesssorKey != nil, we replace {successorAlias}.
+		// If SucessorAlias is "", it's fine. We'll replace {successorAlias} with "".
+		command = strings.Replace(command, "{successorAlias}", topologyRecovery.SuccessorAlias, -1)
```
Comment (shlomi-noach) [inline go/logic/topology_recovery.go]: Please add hint for `{successorAlias}` in `config.go` comments; follow `{successorHost}`'s example

**Comment by shlomi-noach** [comment]
This generally looks :+1: , but do let's solve the name collision by changing to `instance_alias` and `InstanceAlias`. Maybe superfluous when in context of `Instance` but given that `alias` is already widely considered to relate to `cluster`s, this will avoid confusion.

File: go/db/db.go (hunk 1 above)
Comment (enisoc) [inline go/db/db.go]: Done.

File: go/db/db.go (hunk 2 above)
Comment (enisoc) [inline go/db/db.go]: Done.

File: go/inst/instance_dao.go (hunk 4 above)
Comment (enisoc) [inline go/inst/instance_dao.go]: Done.

File: go/logic/topology_recovery.go (hunk 5 above)
Comment (enisoc) [inline go/logic/topology_recovery.go]: Done.

File: go/inst/instance_dao.go (hunk 3 above)
Comment (enisoc) [inline go/inst/instance_dao.go]: Done.

**Comment by enisoc** [comment]
I think I've addressed all the comments. Please take another look. I've also removed the WIP label as I consider it ready whenever you agree.

**Comment by shlomi-noach** [comment]
Gah. Now thats I've merged https://github.com/outbrain/orchestrator/pull/187 this PR has conflicts. Would you please look into?

**Comment by shlomi-noach** [comment]
Naturally it is my fault, as I've asked you to always append to end, and there's a collision with the last comma.
Otherwise this PR looks good and good to merge

**Comment by enisoc** [comment]
I've resolved the merge conflicts.

**Pull request merged**